matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/session/pushrules/EventMatchCondition.kt

```diff
@@ -39,8 +39,8 @@ class EventMatchCondition(
     override fun technicalDescription() = "'$key' matches '$pattern'"
 
     fun isSatisfied(event: Event): Boolean {
-        // TODO encrypted events?
-        val rawJson = MoshiProvider.providesMoshi().adapter(Event::class.java).toJsonValue(event) as? Map<*, *>
+        val rawJson: Map<*, *> = event.mxDecryptionResult?.payload
```

Comment:
This doesn't looks correct to me. The key "path" is based on the full event content.
If you only take the payload it won't work as expected.
Eg the key `content.body` won't work because your raw json will be
```
{ "body": "xxxx" }
```
Instead of 
```
{ "content" : {{ "body": "xxxx" }} "
```

What you would need to do is to take the rawJson, then in case of encrypted event (with a decryption result) you replace the `type`and `content` with what is in event.mxDecryptionResult.

There is an existing test class for this condition you can add test to check that it's working as expected

Reply:
I updated with your suggestion, I replace all the entries from the encrypted raw json with the decrypted one. Also, I had to ensure the event is decrypted before applying the push rules. It was not necessary when I tested last time but it seems that the event is not always decrypted at this moment (it was not the case currently).